cpp/src/arrow/scalar_test.cc

```diff
@@ -297,6 +299,26 @@ TYPED_TEST(TestNumericScalar, MakeScalar) {
   AssertParseScalar(type, "3", ScalarType(3));
 }
 
+template <typename ARROW_TYPE>
+auto GetFloat(double d) {
```

Comment:
Same suggestions as in the other test module.
Also, perhaps this can be factored into one of the testing headers?